server/src/test/java/io/crate/metadata/ReferenceTest.java

```diff
@@ -196,6 +194,42 @@ public void test_mapping_generation_for_float_without_doc_values() throws Except
         assertThat(columnMapping(sourceAsMap, "properties.xs")).isEqualTo(mapping);
     }
 
+    @Test
+    public void test_streaming_column_policy_removed_on_or_after_5_10_0() throws IOException {
+        RelationName relationName = new RelationName("doc", "test");
+        ReferenceIdent referenceIdent = new ReferenceIdent(relationName, "o");
+        var dataType = new ArrayType<>(ObjectType.of(ColumnPolicy.IGNORED)
+            .setInnerType("o2", ObjectType.of(ColumnPolicy.STRICT).build())
+            .build());
+        SimpleReference reference = new SimpleReference(
+            referenceIdent,
+            RowGranularity.DOC,
+            dataType,
+            IndexType.FULLTEXT,
+            false,
+            true,
+            0,
+            COLUMN_OID_UNASSIGNED,
+            false,
+            null
+        );
+        // column policy cannot be recovered so IGNORED and STRICT are changed to DYNAMIC
+        var expectedType = new ArrayType<>(ObjectType.of(ColumnPolicy.DYNAMIC)
+            .setInnerType("o2", ObjectType.of(ColumnPolicy.DYNAMIC).build())
+            .build());
+        SimpleReference expectedRef = (SimpleReference) reference.withValueType(expectedType);
```

Comment:
Wouldn't this be wrong? Can't we use the `columnPolicy` streamed on the `Reference` in such cases?

Reply:
Thank you, the only problem with that approach was inner object policies from nested object types cannot be recovered. I can make the change so that at least the top level object's policy is recovered.